ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1170,15 +1190,27 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager, reconcileConcu
 func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *rayv1.RayCluster, reconcileErr error) (*rayv1.RayCluster, error) {
```

Comment:
Could we avoid handing the special case for `errSuspendCondition`? Instead, we can infer the values of suspending, suspended, or resuming.

* `suspending`
  * false by default
  * false -> true: `suspended` is false and `spec.suspend` is true, and clean up related fields in `status`.
  * true -> false: all Pods are deleted, and set `suspended` from false to true.
* `suspended`
  *  false by default
  * false -> true: when `suspending` transitions from true to false
  * true -> false: `spec.Suspend` is false